### PR TITLE
WIP: Feat/registration

### PR DESCRIPTION
**Pull request by hcnimi**
These will be converted to modals and the corresponding server/db code will be written.

### CODE REVIEW

File: src/client/components/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { updateEmail, updateHash } from '../actions/userInfoActions';
+import { connect } from 'react-redux';
+
+export class Login extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  handleSubmit(formData) {
```
Comment (skoshkarli) [inline src/client/components/Login.jsx]: not doing anything yet?

File: src/client/components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  render() {
+    return (
+      <Grid verticalAlign='middle'>
+        <Grid.Column style={{ maxWidth: 450 }}>
+          <Form
```
Comment (skoshkarli) [inline src/client/components/Login.jsx]: didn't wanna use redux form?

File: src/client/components/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import { updateFirstName, updateLastName, updateEmail, updateHash, updateGender, updateZip, updateWorkZip, updateBirthDate } from '../actions/userInfoActions';
+
+export class Signup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(formData) {
+    alert(JSON.stringify(formData));
```
Comment (skoshkarli) [inline src/client/components/Signup.jsx]: alert?

File: src/client/components/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    const options = [
+      { key: 'm', text: 'Male', value: 'Male' },
+      { key: 'f', text: 'Female', value: 'Female' },
+      { key: 'b', text: 'Both', value: 'Both' }
+    ];
+
+    return (
+      <Grid verticalAlign='middle'>
+        <Grid.Column style={{ maxWidth: 550 }}>
+          <Form
```
Comment (skoshkarli) [inline src/client/components/Signup.jsx]: same as above, dont wanna use redux form?

File: src/client/components/Signup.jsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                placeholder="Work Zip" required
+                errorLabel={ <Label color="red" pointing/> }
+                validationErrors={{
+                  isDefaultRequiredValue: 'Work zip code is required to make location-based recommendations'
+                }}
+              />
+            </Form.Group>
+            <Form.Checkbox
+              name="terms"
+              label="I agree to the Terms and Conditions"
```
Comment (skoshkarli) [inline src/client/components/Signup.jsx]: what are the terms? 😉

File: src/client/components/Signup.jsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            />
+            <Form.Group>
+              <Form.Button content="Submit" color="green"/>
+              <Form.Button type="button" content="Reset" onClick={ () => this.form.reset() }/>
```
Comment (skoshkarli) [inline src/client/components/Signup.jsx]: i dont think we need a reset on a signup form. seems odd. @happysituations @neimanchen  thoughts?

File: src/client/index.jsx [hunk 7]
```diff
@@ -50,6 +51,9 @@ class App extends React.Component {
         <Router>
           <Provider store={store}>
             <div className="content container">
+              {/*<DefaultLayout path="/" component={ClosetBoard} store={store} />*/}
```
Comment (skoshkarli) [inline src/client/index.jsx]: checkout router protected/public routing stuff: i.e https://tylermcginnis.com/react-router-protected-routes-authentication/

File: src/server/index.js [hunk 8]
```diff
@@ -103,24 +108,46 @@ function sendWeather(locationKey, res) {
...
+    userData.password = hash;
+  });
+  db.checkUserExists(userData.email, (err, result) => {
+    if (err) {
+      res.redirect(500, '/signup');
+    }
+    if (result.length) {
+      res.status(500).send('username already exists!');
+    } else {
+      db.createUser(userData, (err, result) => {
```
Comment (skoshkarli) [inline src/server/index.js]: userData is coming directly from req.body. do we need to sanitize?

File: src/client/index.jsx (hunk 7 above)
Comment (hcnimi) [reply src/client/index.jsx]: will look into implementing this. thanks.

File: src/server/index.js (hunk 8 above)
Comment (skoshkarli) [reply src/server/index.js]: can we create an issue for it or do it as part of this pr so we dont forget please

Comment (hcnimi) [reply src/server/index.js]: sure.

**Pull request merged**